app/server/lib/ActiveDoc.ts

```diff
@@ -2496,6 +2497,23 @@ export class ActiveDoc extends EventEmitter {
     }
   }
 
+  private _logSnapshotProgress(docSession: OptDocSession) {
+    const snapshotProgress = this._docManager.storageManager.getSnapshotProgress(this.docName);
+    const lastWindowTime = (snapshotProgress.lastWindowStartedAt &&
+        snapshotProgress.lastWindowDoneAt &&
+        snapshotProgress.lastWindowDoneAt > snapshotProgress.lastWindowStartedAt) ?
+        snapshotProgress.lastWindowDoneAt : Date.now();
```

Comment:
Hmm, the conditional can be false (e.g. it is perfectly possible for the window times not to be set at all yet) so the fallback seems important? Not sure I'm getting it.

Reply:
But `lastWindowTime` is every used only if `lastWindowStartedAt` is filled. But it will get the default value only if `lastWindowStartedAt` is not filled.

Reply:
Ahh I see what you are saying.

I think `lastWindowTime` can get the default value even if `lastWindowStarted` is filled, or at least that was the intention.

Consider the case where `lastWindowStartedAt` is filled, but `lastWindowDoneAt` is not yet filled. Then `lastWindowTime` will be `Date.now()` (basically saying "what if the window got done now or real soon") and the `delay` calculation will use that time. This is so if the window never ends, we'll eventually be logging a very big delay, and alerting can trigger on that.

Reply:
Ahh 😵‍💫 You are right. Boolean logic is sometimes hard :)